app/liquid_tags/unified_embed/tag.rb

```diff
@@ -24,31 +25,28 @@ class Tag < LiquidTagBase
     def self.new(tag_name, input, parse_context)
       stripped_input = ActionController::Base.helpers.strip_tags(input).strip
 
-      # Extract just the URL from the input, without any params, for validation
-      actual_link = extract_only_url(stripped_input)
-
-      # When Listings are disabled, it makes little sense to perform a validate_link
+      # when Listings are disabled, it makes little sense to perform a validate_link
       # network call.
-      handle_listings_disabled!(actual_link)
+      handle_listings_disabled!(stripped_input)
 
       # Before matching against the embed registry, we check if the link
       # is valid (e.g. no typos).
       # If the link is invalid, we raise an error encouraging the user to
       # check their link and try again.
-      validate_link!(actual_link)
-      klass = UnifiedEmbed::Registry.find_liquid_tag_for(link: stripped_input)
+      validated_link = validate_link(stripped_input)
+      klass = UnifiedEmbed::Registry.find_liquid_tag_for(link: validated_link)
 
-      # If there are no OG metatags, we shall render an A-tag. Since the link
-      # has been validated, at least this A-tag will not 404.
-      #
       # Why the __send__?  Because a LiquidTagBase class "privatizes"
       # the `.new` method.  And we want to instantiate the specific
       # liquid tag for the given link.
-      klass.__send__(:new, tag_name, stripped_input, parse_context)
+      klass.__send__(:new, tag_name, validated_link, parse_context)
```

Comment:
so this one line - I can see why we want to remove the params from the head request during validation, but doesn't the embed itself need the url (including params) as passed by the user?

My main consideration would be sites that still use params like `?post_id=1` or the like to select content at request time, rather than urls like `/posts/1` that you expect now. 

Another example, it looks like youtube embeds accept a time parameter, which I assume we need to preserve when we make the tag, even if we ignored it when we validated?